### PR TITLE
Issue #1329 - conditional reference support for transaction bundles

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

**Review by lmsurpre — APPROVED** [review]
LGTM

**Review by prb112 — APPROVED** [review]

**Review by tbieste — APPROVED** [review]
LGTM

File: fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java [hunk 1]
```diff
@@ -2317,9 +2382,9 @@ private static String getUrlPath(Entry entry) {
      * @return local reference map
      */
     private Map<String, String> buildLocalRefMap(Bundle requestBundle, Map<Integer, Entry> validationResponseEntries) throws Exception {
-            Map<String, String> localRefMap = new HashMap<>();
+        Map<String, String> localRefMap = new HashMap<>();
 
-        for (int entryIndex=0; entryIndex<requestBundle.getEntry().size(); ++entryIndex) {
+        for (int entryIndex = 0; entryIndex < requestBundle.getEntry().size(); entryIndex++) {
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java:2387]: so did we have an off-by-one error before?

Comment (lmsurpre) [reply fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java:2387]: i guess either one is fine, so its really just a style thing

Comment (JohnTimm) [reply fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java:2387]: I just updated it for consistency with the coding style of the rest of the class.

**Pull request merged**